homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,177 @@
+"""Config flow for Samsung TV."""
+import socket
+
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, UnhandledResponse
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_ID, CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.components.ssdp import (
+    ATTR_HOST,
+    ATTR_NAME,
+    ATTR_MODEL_NAME,
+    ATTR_MANUFACTURER,
+    ATTR_UDN,
+)
+
+from .const import CONF_MANUFACTURER, CONF_MODEL, DOMAIN, LOGGER, METHODS
+
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str, vol.Required(CONF_NAME): str})
+
+RESULT_AUTH_MISSING = "auth_missing"
+RESULT_SUCCESS = "success"
+RESULT_NOT_FOUND = "not_found"
+RESULT_NOT_SUPPORTED = "not_supported"
+
+
+def _get_ip(host):
+    if host is None:
+        return
+    return socket.gethostbyname(host)
+
+
+async def _async_try_connect(host, name):
+    """Try to connect and check auth."""
+    for method in METHODS:
+        config = {
+            "name": "HomeAssistant",
+            "description": name,
+            "id": "ha.component.samsung",
+            "host": host,
+            "method": method,
+            "port": None,
+            "timeout": None,
+        }
+        try:
+            LOGGER.debug("Try config: %s", config)
+            with Remote(config.copy()):
```

Comment:
If this does sync I/O we need to schedule it on the executor thread pool. Use `await hass.async_add_executor_job(...)`.

Reply:
Perhaps make this function sync and schedule the whole function on the executor? We like to avoid many context switches.